ie/source/Logger.cpp

```diff
@@ -8,16 +8,22 @@
 /**
  * Logger:::Logger
  */
-Logger::Logger(Level level, const std::wstring& filename)
+Logger::Logger(Level level, const std::wstring& filename, const std::wstring& bgfilename,
+                            const std::wstring& fgfilename, const std::wstring& sysfilename,
+                            const std::wstring& tablogfolder)
     : m_level(level), 
-      m_filename(filename)
+      m_filename(filename),
+      m_bgfilename(bgfilename),
+      m_fgfilename(fgfilename),
+      m_sysfilename(sysfilename),
+      m_tablogfolder(tablogfolder)
 {
-    if (m_filename == L"") {
+    if (m_filename == L"" || m_bgfilename == L"" || m_fgfilename == L"" || m_sysfilename == L"" || m_tablogfolder == L"") {

```

Comment:
The way how its implemented it seems we are not allowing only partial logging? For example only BG+FG?